pkg/ovalutil/dpkg.go

```diff
@@ -0,0 +1,135 @@
+package ovalutil
+
+import (
+	"context"
+
+	"github.com/quay/goval-parser/oval"
+	"github.com/rs/zerolog"
+
+	"github.com/quay/claircore"
+)
+
+// DpkgDefsToVulns iterates over the definitions in an oval root and assumes DpkgInfo objects and states.
+//
+// Each Criterion encountered with an EVR string will be translated into a claircore.Vulnerability
+func DpkgDefsToVulns(ctx context.Context, root *oval.Root, protoVulns ProtoVulnsFunc) ([]*claircore.Vulnerability, error) {
+	log := zerolog.Ctx(ctx).With().
+		Str("component", "ovalutil/DpkgDefsToVulns").
+		Logger()
+	ctx = log.WithContext(ctx)
+	vulns := make([]*claircore.Vulnerability, 0, 10000)
+	pkgcache := map[string]*claircore.Package{}
+	cris := []*oval.Criterion{}
+	for _, def := range root.Definitions.Definitions {
+		// create our prototype vulnerability
+		protoVulns, err := protoVulns(def)
+		if err != nil {
+			log.Debug().
+				Err(err).
+				Str("def_id", def.ID).
+				Msg("could not create prototype vulnerabilities")
+			continue
+		}
+		// recursively collect criterions for this definition
+		cris := cris[:0]
+		walkCriterion(ctx, &def.Criteria, &cris)
+		// unpack criterions into vulnerabilities
+		for _, criterion := range cris {
+			// lookup test
+			_, index, err := root.Tests.Lookup(criterion.TestRef)
+			if err != nil {
+				log.Debug().Str("test_ref", criterion.TestRef).Msg("test ref lookup failure. moving to next criterion")
+				continue
+			}
+			test := root.Tests.DpkgInfoTests[index]
+			if len(test.ObjectRefs) == 1 && len(test.StateRefs) == 0 {
+				// We always take an object reference to imply the existence of
+				// that object, so just skip tests with a single object reference
+				// and no associated state object.
+				continue
+			}
+			if len(test.ObjectRefs) != len(test.StateRefs) {
+				log.Debug().Str("test_ref", criterion.TestRef).Msg("object refs and state refs are not in pairs. moving to next criterion")
+				continue
+			}
+			// look at each object,state pair the test references
+			// and create a vuln if an evr tag if found
```

Comment:
```suggestion
			// and create a vuln if an evr tag is found
```

This typo probably also exists on the RPM version :(

Reply:
fixed in both.